### PR TITLE
DD-383 - Moving itinerary docs against delegation instead of delegate

### PR DESCRIPTION
**Pull request by nboyse**

### CODE REVIEW

File: dso/services/delegate.py [hunk 1]
```diff
@@ -440,11 +436,41 @@ def get_vip_document_count(delegate: Delegate) -> int:
...
     return [doc for doc in docs]
 
 
+@transaction.atomic
+def set_itinerary_document(
+    user: User,
+    delegation: Delegation,
+    document_ids: List[str],
+):
+    logger.debug(f"setting itinerary document for delegation {delegation}")
```
Comment (mikeylockwood) [inline dso/services/delegate.py:452]: Is the logging level for this logger set to debug in production or is this just for development?

File: dso/services/delegate.py [hunk 2]
```diff
@@ -440,11 +436,41 @@ def get_vip_document_count(delegate: Delegate) -> int:
 
 
 def get_itinerary_documents(delegate: Delegate):
-    docs = delegate.itinerary_documents.all().order_by("display_name")
+    docs = []
+    if delegate.delegation:
+        docs = delegate.delegation.itinerary_documents.all().order_by("display_name")
```
Comment (mikeylockwood) [inline dso/services/delegate.py:441]: I don't think you need the `.all()` if you are calling `.order_by()`

File: dso/services/delegate.py [hunk 3]
```diff
@@ -440,11 +436,41 @@ def get_vip_document_count(delegate: Delegate) -> int:
...
+):
+    logger.debug(f"setting itinerary document for delegation {delegation}")
+
+    logger.debug(document_ids)
+
+    delegation.updated_by = user.username
+
+    delegation.itinerary_documents.clear()
+
+    if document_ids:
```
Comment (mikeylockwood) [inline dso/services/delegate.py:460]: If `document_ids` is always a list, this `if` is not necessary

File: dso/migrations/0150_migrate_itinerary_to_delegation.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
+from django.db import migrations, models
+
+
+def map_itinerary_docs_to_delegation_from_delegate(apps, schema_editor):
+    DelegateModel = apps.get_model("dso", "delegate")
+
+    for delegate in DelegateModel.objects.all():
+        if delegate.itinerary_documents.all():
```
Comment (mikeylockwood) [inline dso/migrations/0150_migrate_itinerary_to_delegation.py:8]: Can remove this `if` as it is just wont loop over the empty queryset

File: dso/views/delegation_readonly.py [hunk 5]
```diff
@@ -211,17 +211,13 @@ def delegation_overview(request, event_slug, delegate_id):  # noqa C901 - Too co
             additional_guests_count += int(delegate.additional_guests)
 
     country_brief = None
+    itinerary_documents = None
     if delegation_model:
         country_brief = get_country_brief_for_event(
             delegation_model.event, delegation_model.country
         )
+        itinerary_documents = get_itinerary_documents(delegation_model.delegate_set.first())
```
Comment (mikeylockwood) [inline dso/views/delegation_readonly.py:219]: The definition for `get_itinerary_documents` doesn't accept an argument

**Review by mikeylockwood — APPROVED** [review]

File: dso/services/delegate.py (hunk 1 above)
Comment (nboyse) [reply dso/services/delegate.py:452]: It'll also be in prod

**Pull request merged**